solver/src/main/java/org/chocosolver/solver/constraints/nary/binPacking/PropBinPacking.java

```diff
@@ -348,6 +336,16 @@ public void propagate(int idxVarInProp, int mask) throws ContradictionException
     @Override
     public void propagate(int evtmask) throws ContradictionException {
         if(PropagatorEventType.isFullPropagation(evtmask)) {
+            for(int j = 0; j<nbAvailableBins; j++) {
+                int pj = 0;
+                for (int i = 0; i < nbItems; i++) {
+                    if (itemBin[i].contains(j + offset)) {
+                        P[j].add(i);
+                        pj += itemSize[i];
+                    }
+                }
+                sumP[j].set(pj);
```

Comment:
now we are within the propagate method and no longer in the constructor, maybe we can reuse the 
```
if(itemBin[i].isInstantiated()) {
                        R[j].add(i);
                        rj += itemSize[i];
                    } 
                    sumP[j].set(rj);
```
associated with the filtering that was missing, if you think that doing so would make the code clearer.
If you prefer keeping it as it is, I am fine with that too

Reply:
No need to, it will be done correctly at line 353